soroban-env-host/src/auth.rs

```diff
@@ -404,6 +411,7 @@ impl AuthorizedFunction {
                             Ok(addr.clone())
                         })?,
                     function_name: contract_fn.function_name.try_into_val(host)?,
+                    // why is this intentionally non-metered? the visit_obj above is metered.
```

Comment:
This whole function is not meant to be metered. I'm not sure if we have a non-metered visit_obj version. 
I think we can just get rid of it (or the current implementation) and replace with `host.with_free_budget(self.to_xdr())`.

Reply:
There is a bit of divergence between the `to_xdr` and `to_xdr_non_metered` with the `exhausted_sub_invocations_only` check in the non-metered version only. I'll leave this piece along and perhaps you can clean it up later.

Reply:
Tracking issue for followup https://github.com/stellar/rs-soroban-env/issues/977